packages/flutter_tools/lib/src/commands/make_host_app_editable.dart

```diff
@@ -10,69 +10,57 @@ import '../runner/flutter_command.dart';
-    addSubcommand(MakeHostAppEditableAndroidCommand());
-    addSubcommand(MakeHostAppEditableIosCommand());
+    requiresPubspecYaml();
+
+    argParser.addFlag(
+      'ios',
+      help: 'Whether to make this project\'s iOS app editable.',
+      negatable: false,
+    );
+    argParser.addFlag(
+      'android',
+      help: 'Whether ot make this project\'s Android app editable.',
+      negatable: false,
+    );
   }
 
-  @override
-  final String name = 'make-host-app-editable';
+  FlutterProject _project;
 
   @override
-  final String description = 'Commands for making host apps editable within a Flutter project';
+  final String name = 'make-host-app-editable';
 
   @override
-  bool get hidden => true;
+  final String description = 'Converts generated host apps to editable, non-generated host apps.\n\n'
+    'Use flags to specify which host app to make editable. If no flags are provided then all host apps will be made editable.\n\n'
+    'Once a host app is made editable, that host app cannot be regenerated by Flutter and it will not receive future template changes.';
 
   @override
-  Future<Null> runCommand() async { }
-}
-
-abstract class MakeHostAppEditableSubCommand extends FlutterCommand {
-  MakeHostAppEditableSubCommand() {
-    requiresPubspecYaml();
-  }
-
-  FlutterProject _project;
-
-  @override
-  @mustCallSuper
-  Future<Null> runCommand() async {
-    await _project.ensureReadyForPlatformSpecificTooling();
-  }
+  bool get hidden => false;
 
   @override
   Future<Null> validateCommand() async {
     await super.validateCommand();
     _project = await FlutterProject.current();
+    // TODO(mattcarroll): change _project.isModule to use Greg's updated inspection
     if (!_project.isModule)
-      throw ToolExit("Only projects created using 'flutter create -t module' can have their host apps made editable.");
+      throw ToolExit("Only projects created using 'flutter create -t application' can have their host apps made editable.");
   }
-}
-
-class MakeHostAppEditableAndroidCommand extends MakeHostAppEditableSubCommand {
-  @override
-  String get name => 'android';
-
-  @override
-  String get description => 'Make an Android host app editable within a Flutter project';
 
   @override
   Future<Null> runCommand() async {
-    await super.runCommand();
-    await _project.android.makeHostAppEditable();
-  }
-}
-
-class MakeHostAppEditableIosCommand extends MakeHostAppEditableSubCommand {
-  @override
-  String get name => 'ios';
-
-  @override
-  String get description => 'Make an iOS host app editable within a Flutter project';
+    await _project.ensureReadyForPlatformSpecificTooling();
 
-  @override
-  Future<Null> runCommand() async {
-    await super.runCommand();
-    await _project.ios.makeHostAppEditable();
+    final bool isAndroidRequested = argResults.arguments.contains('android');
+    final bool isIOSRequested = argResults.arguments.contains('ios');
+
+    if (isAndroidRequested) {
```

Comment:
This logic will do nothing if a user decides to pass both `--ios` and `--android`, right?  Seems like we might need to check for the case where both are specified explicitly.

Reply:
It would do something, but not the right thing.  I've updated the logic.

Reply:
Should `if (isAndroidRequested == isIOSRequested) {` be `if (isAndroidRequested == false && isIOSRequested == false) {`?  Or does it not hurt anything to run these commands multiple times (they immediately exit if already done)?